### PR TITLE
Fix Error: Typed property MatchBot\Domain\SalesforceProxyRepository::…

### PR DESCRIPTION
**Pull request by bdsl**
…$logger must not be accessed before initialization

The EntityManager does not set the logger in repositories, so when we need a repository we have to take it from the DI container which does inject the logger, not the EM.

### CODE REVIEW

File: src/Application/Actions/Donations/Create.php [hunk 1]
```diff
@@ -149,9 +149,7 @@ protected function action(Request $request, Response $response, array $args): Re
         if ($donation->getPsp() === 'stripe') {
             if (empty($donation->getCampaign()->getCharity()->getStripeAccountId())) {
                 // Try re-pulling in case charity has very recently onboarded with for Stripe.
-                $repository = $this->entityManager->getRepository(Campaign::class);
```
Comment (bdsl) [inline src/Application/Actions/Donations/Create.php:152]: entityManager doesn't know that our repos need to have a logger

File: src/Domain/SalesforceProxyRepository.php [hunk 2]
```diff
@@ -18,35 +18,17 @@
...
     protected function logError(string $message): void
     {
-        if ($log = $this->getLogger()) {
-            $log->error($message);
-            return;
-        }
-
-        // Fallback if logger not configured
-        echo "ERROR: $message";
+        $this->logger->error($message);
```
Comment (bdsl) [inline src/Domain/SalesforceProxyRepository.php:21]: the logger property is already typed as Logger so if it isn't there getLogger will just crash with a TypeError it won't return a falsy value. So all this stuff is redundant.

File: src/Domain/SalesforceProxyRepository.php [hunk 3]
```diff
@@ -67,9 +49,4 @@ protected function getClient(): Client\Common
 
         return $this->client;
     }
-
-    protected function getLogger(): ?LoggerInterface
```
Comment (bdsl) [inline src/Domain/SalesforceProxyRepository.php:71]: property was typed as non-nullable so no reason to make return it as nullable here. And then a protected getter for a protected property is redundant.

File: tests/TestCase.php [hunk 4]
```diff
@@ -27,12 +27,22 @@ class TestCase extends PHPUnitTestCase
...
+     */
+    private array $appInstance = [0 => null, 1 => null];
+
     /**
      * @return App
      * @throws Exception
      */
     protected function getAppInstance(bool $withRealRedis = false): App
     {
+        $memoizedInstance = $this->appInstance[(int)$withRealRedis];
```
Comment (bdsl) [inline tests/TestCase.php:41]: Having added a call to getAppInstance in the CreateTest setup there are now some duplicate calls to this function form within the same test. We have to return the same instance in that case.

At the end of each test PHPUnit throws away the test case object so the memoized instance(s) get garbaged.

**Review by NoelLH — COMMENTED** [review]
Looks good!

**Pull request merged**